### PR TITLE
feat: Add HTTP client customization support

### PR DESCRIPTION
**Pull request by dasheck0**
## Summary

This pull request introduces HTTP client customization support to the PAYONE Commerce Platform Python SDK, allowing developers to configure the underlying `httpx.AsyncClient` globally or for specific API clients. This feature replicates the functionality from the Java SDK while following Python best practices.

### HTTP Client Customization
- **Global HTTP Client Configuration**: Added support for customizing the `httpx.AsyncClient` globally via `CommunicatorConfiguration`
- **Client-Specific HTTP Client Configuration**: Added support for setting custom HTTP clients on individual API clients via `set_http_client()` method
- **Priority Logic**: Implemented clear priority order where client-specific settings override global settings, which override default settings

### Implementation Details
- **Enhanced `CommunicatorConfiguration`**: Added optional `http_client` parameter to constructor and corresponding getter/setter methods
- **Enhanced `BaseApiClient`**: Added client-specific HTTP client support with `set_http_client()` and `get_http_client()` methods
- **Priority Order**: Client-specific HTTP client → Global HTTP client → Default HTTP client

### Global HTTP Client Configuration
```python
import httpx
from pcp_serversdk_python import CommunicatorConfiguration
from pcp_serversdk_python.endpoints import CommerceCaseApiClient

# Create a custom HTTP client with specific configuration
custom_http_client = httpx.AsyncClient(
    timeout=httpx.Timeout(30.0),  # 30 second timeout
    limits=httpx.Limits(max_connections=100, max_keepalive_connections=20),
    headers={"User-Agent": "MyApp/1.0"}
)

# Create configuration with custom HTTP client
config = CommunicatorConfiguration(
    api_key="your_api_key",
    api_secret="your_api_secret", 
    host="https://api.preprod.commerce.payone.com",
    http_client=custom_http_client
)

# All API clients created with this config will use the custom HTTP client
commerce_case_client = CommerceCaseApiClient(config)
```

### Client-Specific HTTP Client Configuration
```python
# Create API clients
commerce_case_client = CommerceCaseApiClient(config)
checkout_client = CheckoutApiClient(config)

# Set custom HTTP client only for the commerce case client
custom_client = httpx.AsyncClient(timeout=httpx.Timeout(60.0))
commerce_case_client.set_http_client(custom_client)

# commerce_case_client will use the custom client
# checkout_client will use the global or default client
```

## Testing

- **17 new test cases** covering all HTTP client customization scenarios
- **All existing tests pass** (91/91) - no breaking changes
- **Comprehensive coverage** of global configuration, client-specific configuration, priority logic, and edge cases
- **Integration tests** verify the feature works with existing functionality

## Documentation

- **Updated README.md** with comprehensive documentation section
- **Code examples** for common use cases (timeouts, headers, proxies)
- **Priority order explanation** with clear examples
- **Working demonstration** added to `example/main.py`

## Common Use Cases

**Setting Custom Timeouts:**
```python
slow_client = httpx.AsyncClient(timeout=httpx.Timeout(120.0))
config = CommunicatorConfiguration("key", "secret", "host", slow_client)
```

**Adding Custom Headers:**
```python
headers = {"X-Custom-Header": "value", "User-Agent": "MyApp/2.0"}
client = httpx.AsyncClient(headers=headers)
config = CommunicatorConfiguration("key", "secret", "host", client)
```

**Configuring Proxy:**
```python
proxy_client = httpx.AsyncClient(proxies="http://proxy.example.com:8080")
config = CommunicatorConfiguration("key", "secret", "host", proxy_client)
```

## Backward Compatibility

✅ **Fully backward compatible** - all existing code continues to work without changes
✅ **Optional parameter** - `http_client` parameter in `CommunicatorConfiguration` is optional
✅ **Default behavior preserved** - when no custom HTTP client is provided, the SDK behaves exactly as before

## Quality Assurance

- ✅ All tests pass (91/91)
- ✅ Code follows project linting standards
- ✅ Comprehensive documentation
- ✅ Working examples provided
- ✅ No breaking changes
- ✅ Follows Python best practices

## Related Issues

This feature addresses the same need as [PAYONE-GmbH/PCP-ServerSDK-java#41](https://github.com/PAYONE-GmbH/PCP-ServerSDK-java/issues/41) and implements the same solution pattern as [PAYONE-GmbH/PCP-ServerSDK-java#42](https://github.com/PAYONE-GmbH/PCP-ServerSDK-java/pull/42), adapted for Python and `httpx.AsyncClient`.

## Review Notes

The implementation follows the exact same pattern as the Java SDK but uses Python-specific best practices:
- Uses `httpx.AsyncClient` instead of `OkHttpClient`
- Follows Python naming conventions
- Maintains async/await patterns
- Includes proper type hints
- Ensures proper resource management

---
Pull Request opened by [Augment Code](https://www.augmentcode.com/) with guidance from the PR author

### CODE REVIEW

File: pcp_serversdk_python/endpoints/BaseApiClient.py [hunk 1]
```diff
@@ -46,17 +46,50 @@ class BaseApiClient:
     CHECKOUT_ID_REQUIRED_ERROR = "Checkout ID is required"
     PAYMENT_INFORMATION_ID_REQUIRED_ERROR = "Payment Information ID is required"
     PAYMENT_EXECUTION_ID_REQUIRED_ERROR = "Payment Execution ID is required"
+    JSON_PARSE_ERROR = "Failed to parse JSON response"
```
Comment (BenchmarkingBuffalo) [inline pcp_serversdk_python/endpoints/BaseApiClient.py:49]: Is this constant used anywhere?

**Review by BenchmarkingBuffalo — APPROVED** [review]
